Spanner/tests/Perf/Operation.php

```diff
@@ -0,0 +1,267 @@
+
+class Operation
+{
+    /**
+     * @var string
+     */
+    private $database;
+
+    /**
+     * @var array
+     */
+    private $parameters;
+
+    /**
+     * float
+     */
+    private $totalWeight;
+
+    /**
+     * @var array
+     */
+    private $weights;
+
+    /**
+     * @var array
+     */
+    private $operations;
+
+    /**
+     * @var array
+     */
+    private $latency;
+
+    /**
+     * @var array
+     */
+    private $keys;
+
+    /**
+     * @param string $database
+     * @param array $parameters
+     * @param float $totalWeight
+     * @param array $weights
+     * @param array $operations
+     * @param array $latency
+     */
+    public function __construct(
+        $database,
+        array $parameters,
+        $totalWeight,
+        array $weights,
+        array $operations,
+        array $latency
+    ) {
+        $this->database = $database;
+        $this->parameters = $parameters;
+        $this->totalWeight = $totalWeight;
+        $this->weights = $weights;
+        $this->operations = $operations;
+        $this->latency = $latency;
+    }
+
+    public function load()
+    {
+        $this->keys = [];
+
+        $startTime = microtime(true);
+        $snapshot = $this->database->snapshot();
+
+        // Kind of assuming that id is always name of PK in whatever table you choose
+        $results = $snapshot->execute('SELECT id FROM ' . $this->parameters['table']);
+        foreach ($results as $row) {
+            $this->keys[] = $row['id'];
+        }
+
+        return microtime(true) - $startTime;
+    }
+
+    /**
+     * Run a single thread of the workload
+     *
+     * @return void
+     */
+    public function run()
+    {
+        if (!is_array($this->keys)) {
+            throw new \RuntimeException('Call Operation::load() first.');
+        }
+
+        $operationCount = (int) $this->parameters['operationcount'];
+        for ($i = 0; $i < $operationCount; $i++) {
+            $weight = (float) rand(0, $this->totalWeight * 10000) / 10000.0;
+
+            for ($j = 0; $j < count($this->weights); $j++) {
+                if ($weight <= $this->weights[$j]) {
+                    $operation = $this->operations[$j];
+
+                    $this->latency[$operation][] = $this->runOperation(
+                        $this->database,
+                        $this->parameters['table'],
+                        $operation
+                    );
+                    break;
+                }
+            }
+        }
+
+        return $this->latency;
+    }
+
+    /**
+     * Executes an operation.
+     *
+     * @param string $database
+     * @param string $table
+     * @param array $operation
+     * @return void
+     */
+    private function runOperation($database, $table, $operation)
+    {
+        $key = $this->keys[array_rand($this->keys)];
+        switch ($operation) {
+            case 'read':
+                $optime = $this->performRead($database, $table, $key);
```

Comment:
We could make this a little more concise with:

```suggestion
                return $this->performRead($database, $table, $key);
```